persistentvolumeclaims.go

```diff
@@ -13,19 +17,167 @@ import (
 )
 
 type pvcresponse struct {
-	Class     string                            `json:"class"`
-	Mode      corev1.PersistentVolumeAccessMode `json:"mode"`
-	Name      string                            `json:"name"`
-	Namespace string                            `json:"namespace"`
-	Labels    map[string]string                 `json:"labels"`
-	Size      resource.Quantity                 `json:"size"`
+	Name      string                              `json:"name"`
+	Namespace string                              `json:"namespace"`
+	Status    pvcStatus                           `json:"status"`
+	Age       string                              `json:"age"`
+	Capacity  resource.Quantity                   `json:"capacity"`
+	Modes     []corev1.PersistentVolumeAccessMode `json:"modes"`
+	Class     string                              `json:"class"`
+	Notebooks []string                            `json:"notebooks"`
+	Labels    map[string]string                   `json:"labels"`
 }
 
 type pvcsresponse struct {
 	APIResponseBase
 	PersistentVolumeClaims []pvcresponse `json:"pvcs"`
 }
 
+type getpvcresponse struct {
+	APIResponseBase
+	Pvc corev1.PersistentVolumeClaim `json:"pvc"`
+}
+
+type pvcpodsresponse struct {
+	APIResponseBase
+	Pods []corev1.Pod `json:"pods"`
+}
+
+type pvceventsresponse struct {
+	APIResponseBase
+	Events []corev1.Event `json:"events"`
+}
+
+// pvcPhase is the phase of a PVC
+type pvcPhase string
+
+// status represents the status of a PVC.
+type pvcStatus struct {
+	Message      string   `json:"message"`
+	Phase        pvcPhase `json:"phase"`
+	State        string   `json:"state"`
+	Key          string   `json:"key"`
+	EventMessage string   `json:"eventMessage"`
+}
+
+const (
+	PvcPhaseReady         pvcPhase = "ready"
+	PvcPhaseWaiting       pvcPhase = "waiting"
+	PvcPhaseWarning       pvcPhase = "warning"
+	PvcPhaseError         pvcPhase = "error"
+	PvcPhaseUninitialized pvcPhase = "uninitialized"
+	PvcPhaseUnavailable   pvcPhase = "unavailable"
+	PvcPhaseTerminating   pvcPhase = "terminating"
+	PvcPhaseStopped       pvcPhase = "stopped"
+)
+
+// Set the status of the pvc
```

Comment:
`// https://github.com/kubeflow/kubeflow/blob/v1.7.0/components/crud-web-apps/volumes/backend/apps/common/status.py#L4`

Reply:
added. Agreed that it can help to know where this comes from